modules/core/src/lib/picking/pick-info.js

```diff
@@ -141,11 +140,20 @@ export function getLayerPickingInfo({layer, info, mode}) {
 }
 
 // Indentifies which viewport, if any corresponds to x and y
+// If multiple viewports contain the target pixel, use priority:
+//  - last viewport that the picked layer was rendered in
+//  - last viewport drawn
 // Returns first viewport if no match
-// TODO - need to determine which viewport we are in
-// TODO - document concept of "primary viewport" that matches all coords?
-// TODO - static method on Viewport class?
-function getViewportFromCoordinates({viewports}) {
-  const viewport = viewports[0];
-  return viewport;
+function getViewportFromCoordinates(viewports, pixel, layer) {
+  const lastDrawnViewport = layer && layer.internalState.viewport;
+  if (lastDrawnViewport && lastDrawnViewport.containsPixel(pixel)) {
```

Comment:
Doesn't this duplicate the logic below? Wouldn't the last drawn viewport always be last in the viewports array?

Reply:
This accounts for layers being hidden in a viewport via `layerFilter`